clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1135,14 +1145,14 @@ public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception
         log.trace("Sending OffsetCommit request with {} to coordinator {}", offsets, coordinator);
 
         return client.send(coordinator, builder)
-                .compose(new OffsetCommitResponseHandler(offsets));
+                .compose(new OffsetCommitResponseHandler(offsets, generation));
```

Comment:
I don't think this generation object is protected by synchronized block.

Reply:
Since we're accessing a local variable, I don't think we need synchronization. Note `generationIfStable` is synchronized though.

Reply:
The generation object is immutable, so as long as we get a reference of it we do not need synchronization anymore.